pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -649,7 +649,7 @@ protected CompletableFuture<Void> doReconsumeLater(Message<?> message, AckType a
                 if (reconsumeTimes > this.deadLetterPolicy.getMaxRedeliverCount()
                         && StringUtils.isNotBlank(deadLetterPolicy.getDeadLetterTopic())) {
                     initDeadLetterProducerIfNeeded();
-                    deadLetterProducer.thenAccept(dlqProducer -> {
+                    deadLetterProducer.thenAcceptAsync(dlqProducer -> {
```

Comment:
It seems can use `internalPinnedExecutor` execution.

Reply:
yes, agree

Reply:
changed to internalPinnedExecutor